.github/workflows/main.yml

```diff
@@ -101,37 +102,34 @@ jobs:
       run: echo CARGO_CONTAINER_TESTS=1 >> $GITHUB_ENV
       if: matrix.os == 'ubuntu-latest'
 
-    - run: cargo test
+    - run: cargo test -p cargo
     - name: Clear intermediate test output
       run: ci/clean-test-output.sh
     - name: gitoxide tests (all git-related tests)
-      run: cargo test git
+      run: cargo test -p cargo git
       env:
         __CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2: 1
       # The testsuite generates a huge amount of data, and fetch-smoke-test was
     # running out of disk space.
     - name: Clear test output
       run: ci/clean-test-output.sh
+    # This only tests `cargo fix` because fix-proxy-mode is one of the most
+    # complicated subprocess management in Cargo.
     - name: Check operability of rustc invocation with argfile
+      run: 'cargo test -p cargo --test testsuite -- fix::'
       env:
         __CARGO_TEST_FORCE_ARGFILE: 1
-      run: |
-        # This only tests `cargo fix` because fix-proxy-mode is one of the most
-        # complicated subprocess management in Cargo.
-        cargo test --test testsuite -- fix::
-    - run: cargo test --manifest-path crates/cargo-test-support/Cargo.toml
-      env:
-        CARGO_TARGET_DIR: target
+    - run: cargo test -p cargo-test-support
     - run: cargo test -p cargo-platform
     - run: cargo test -p cargo-util
-    - run: cargo test --manifest-path crates/home/Cargo.toml
-    - run: cargo test --manifest-path crates/mdman/Cargo.toml
-    - run: cargo build --manifest-path crates/credential/cargo-credential-1password/Cargo.toml
-    - run: cargo build --manifest-path crates/credential/cargo-credential-gnome-secret/Cargo.toml
+    - run: cargo test -p home
+    - run: cargo test -p mdman
+    - run: cargo build -p cargo-credential-1password
+    - run: cargo build -p cargo-credential-gnome-secret
       if: matrix.os == 'ubuntu-latest'
-    - run: cargo build --manifest-path crates/credential/cargo-credential-macos-keychain/Cargo.toml
+    - run: cargo build -p cargo-credential-macos-keychain
       if: matrix.os == 'macos-latest'
-    - run: cargo build --manifest-path crates/credential/cargo-credential-wincred/Cargo.toml
+    - run: cargo build -p cargo-credential-wincred
```

Comment:
Can we make it work so we just do `cargo test --all` or `cargo build --all`?

Reply:
No, it will fail to build `cargo-credential-wincred` on macOS or else. See similar feature request <https://github.com/rust-lang/cargo/issues/5220> and linked issues there.

Reply:
This is addressable.  Not doing so negates a lot of the value add from having a workspace (scaling up our processes)

Now, whether we defer this out, that is another thing.  At this point though, we are deferring out a lot and I'd like a list of what all is deferred out and an idea of how immediately we plan to address it or what mechanism we are using to track it.

Reply:
Just put them under “Unresolved issues” section in the PR description. Thank you for the advice!